### PR TITLE
Screenshot preview

### PR DESCRIPTION
**Pull request by john-rocky**
A preview is now displayed when sharing screenshots.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Minor updates to the YOLO iOS app README and Xcode project files for better model handling and development experience.

### 📊 Key Changes
- **Formatting Adjustments:** Improved markdown formatting in `README.md` for better code readability.
- **Development Team:** Removed fixed Apple Developer Team IDs in project settings, adding flexibility for different developers.
- **Code Cleanup:** Removed unused YOLO model references from the Xcode project file.
- **New Xcode Scheme:** Added a shared Xcode scheme for easier project setup and building.
- **UI Enhancements:** Added a function for sharing images with a customizable alert.
- **Bug Fixes:** Addressed minor bugs and code inconsistencies in various Swift files.

### 🎯 Purpose & Impact
- **Enhanced Readability:** The cleaned-up markdown and code blocks make it easier for new developers to follow instructions.
- **Developer Flexibility:** Removing fixed developer team IDs allows multiple developers to work on the project without manual changes.
- **Improved Development Workflow:** The new shared Xcode scheme simplifies the build and test process.
- **User Experience:** UI enhancements provide a better interface for sharing images and screenshots, improving overall usability.

### CODE REVIEW

**Review by ambitious-octopus — COMMENTED** [review]
Thanks for creating a new PR @john-rocky.

File: YOLO.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -35,11 +30,6 @@
...
 		636EFCB821E62E3900DE43BC /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
-		6381D2132B7817C200ABA4E8 /* yolov8l.mlpackage */ = {isa = PBXFileReference; lastKnownFileType = folder.mlpackage; path = yolov8l.mlpackage; sourceTree = "<group>"; };
-		6381D2142B7817C200ABA4E8 /* yolov8x.mlpackage */ = {isa = PBXFileReference; lastKnownFileType = folder.mlpackage; path = yolov8x.mlpackage; sourceTree = "<group>"; };
-		6381D2152B7817C200ABA4E8 /* yolov8s.mlpackage */ = {isa = PBXFileReference; lastKnownFileType = folder.mlpackage; path = yolov8s.mlpackage; sourceTree = "<group>"; };
-		6381D2162B7817C200ABA4E8 /* yolov8m.mlpackage */ = {isa = PBXFileReference; lastKnownFileType = folder.mlpackage; path = yolov8m.mlpackage; sourceTree = "<group>"; };
-		6381D2172B7817C200ABA4E8 /* yolov8n.mlpackage */ = {isa = PBXFileReference; lastKnownFileType = folder.mlpackage; path = yolov8n.mlpackage; sourceTree = "<group>"; };
```
Comment (ambitious-octopus) [inline YOLO.xcodeproj/project.pbxproj:42]: Also here!

File: YOLO.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -87,11 +77,6 @@
 		63A946D8271800E20001C3ED /* Models */ = {
 			isa = PBXGroup;
 			children = (
-				6381D2132B7817C200ABA4E8 /* yolov8l.mlpackage */,
-				6381D2162B7817C200ABA4E8 /* yolov8m.mlpackage */,
-				6381D2172B7817C200ABA4E8 /* yolov8n.mlpackage */,
-				6381D2152B7817C200ABA4E8 /* yolov8s.mlpackage */,
-				6381D2142B7817C200ABA4E8 /* yolov8x.mlpackage */,
```
Comment (ambitious-octopus) [inline YOLO.xcodeproj/project.pbxproj:94]: and here too

File: YOLO.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -379,7 +359,7 @@
 				ASSETCATALOG_COMPILER_INCLUDE_ALL_APPICON_ASSETS = NO;
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 0;
-				DEVELOPMENT_TEAM = 3MR4P6CL3X;
+				DEVELOPMENT_TEAM = "";
```
Comment (ambitious-octopus) [inline YOLO.xcodeproj/project.pbxproj:362]: why this change?

File: YOLO/ViewController.swift [hunk 4]
```diff
@@ -558,7 +596,16 @@ extension ViewController: AVCapturePhotoCaptureDelegate {
         if let dataImage = photo.fileDataRepresentation() {
             let dataProvider = CGDataProvider(data: dataImage as CFData)
             let cgImageRef: CGImage! = CGImage(jpegDataProviderSource: dataProvider!, decode: nil, shouldInterpolate: true, intent: .defaultIntent)
-            let image = UIImage(cgImage: cgImageRef, scale: 0.5, orientation: UIImage.Orientation.right)
+            var orientation = UIImage.Orientation.right
```
Comment (ambitious-octopus) [inline YOLO/ViewController.swift:599]: Am I wrong or is this change related to landscape mode? Maybe we need to merge the branch landscape first! @john-rocky What do you think about it?

**Review by ambitious-octopus — CHANGES_REQUESTED** [review]

File: YOLO.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -13,11 +13,6 @@
 		636EFCAF21E62DD300DE43BC /* VideoCapture.swift in Sources */ = {isa = PBXBuildFile; fileRef = 636EFCA221E62DD300DE43BC /* VideoCapture.swift */; };
 		636EFCB321E62DD300DE43BC /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 636EFCA721E62DD300DE43BC /* AppDelegate.swift */; };
 		636EFCB921E62E3900DE43BC /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 636EFCB821E62E3900DE43BC /* Assets.xcassets */; };
-		6381D2182B7817C200ABA4E8 /* yolov8l.mlpackage in Sources */ = {isa = PBXBuildFile; fileRef = 6381D2132B7817C200ABA4E8 /* yolov8l.mlpackage */; };
-		6381D2192B7817C200ABA4E8 /* yolov8x.mlpackage in Sources */ = {isa = PBXBuildFile; fileRef = 6381D2142B7817C200ABA4E8 /* yolov8x.mlpackage */; };
-		6381D21A2B7817C200ABA4E8 /* yolov8s.mlpackage in Sources */ = {isa = PBXBuildFile; fileRef = 6381D2152B7817C200ABA4E8 /* yolov8s.mlpackage */; };
-		6381D21B2B7817C200ABA4E8 /* yolov8m.mlpackage in Sources */ = {isa = PBXBuildFile; fileRef = 6381D2162B7817C200ABA4E8 /* yolov8m.mlpackage */; };
-		6381D21C2B7817C200ABA4E8 /* yolov8n.mlpackage in Sources */ = {isa = PBXBuildFile; fileRef = 6381D2172B7817C200ABA4E8 /* yolov8n.mlpackage */; };
```
[the remark this answers is not in the document]
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj:20]: In my opinion, these are just references to mlpackage files, and I thought it would be better if they were not there.
If you need them, please let me know and I will restore them and recommit them.
Github does not allow files larger than 100mb in total, so only the mlpackage reference records remain. From experience, I thought that these could cause conflicts when users try this project locally, so I removed them.
Alternatively, we may want to include *.mlpackage in our .gitignore.

File: YOLO.xcodeproj/project.pbxproj (hunk 1 above)
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj:42]: Same reason as above. If you want to restore it, I will.

File: YOLO.xcodeproj/project.pbxproj (hunk 2 above)
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj:94]: Same reason as above. If you want to restore it, I will.

File: YOLO.xcodeproj/project.pbxproj (hunk 3 above)
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj:362]: I left it blank because when users build the project locally, they need to change it to their own team. If you want to change it back to the original team, please let me know.

File: YOLO/ViewController.swift (hunk 4 above)
Comment (john-rocky) [reply YOLO/ViewController.swift:599]: This is indeed code that supports landscape mode, and it will still work in portrait mode, but I can change it back to the original code if necessary, or I can commit landscape mode to reflect that.

File: YOLO.xcodeproj/project.pbxproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj:20]: スクリーンショット 2024-07-26 8 28 16

**Review by ambitious-octopus — COMMENTED** [review]
Hey @john-rocky, this is the next PR to integrate, can you resolve conflicts?

**Comment by john-rocky** [comment]
@ambitious-octopus 
I have resolved these conflicts.
This feature works with the landscape mode.

File: YOLO.xcodeproj/project.pbxproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (ambitious-octopus) [reply YOLO.xcodeproj/project.pbxproj:20]: @john-rocky Let's keep these references so we can easily point out where to place the models during the build phase.

Also, could you add a brief paragraph in the README to clarify that these are just references?

https://github.com/ultralytics/yolo-ios-app/blob/494eedac0528178422cb0e9ab9992c083969e679/README.md?plain=1#L63-L77

File: YOLO.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ambitious-octopus) [reply YOLO.xcodeproj/project.pbxproj:41]: Also here, keep these references!

File: YOLO.xcodeproj/project.pbxproj (hunk 2 above)
Comment (ambitious-octopus) [reply YOLO.xcodeproj/project.pbxproj:93]: Keep also here!

File: YOLO.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -209,13 +194,8 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				6381D21B2B7817C200ABA4E8 /* yolov8m.mlpackage in Sources */,
-				6381D21C2B7817C200ABA4E8 /* yolov8n.mlpackage in Sources */,
 				636EFCAF21E62DD300DE43BC /* VideoCapture.swift in Sources */,
```
Comment (ambitious-octopus) [inline YOLO.xcodeproj/project.pbxproj:197]: keep these references

File: YOLO.xcodeproj/project.pbxproj (hunk 3 above)
Comment (ambitious-octopus) [reply YOLO.xcodeproj/project.pbxproj:361]:
```suggestion
				DEVELOPMENT_TEAM = 3MR4P6CL3X;
```
Keep the Development team!

**Review by ambitious-octopus — CHANGES_REQUESTED** [review]
Hey @john-rocky, please keep the references and `DEVELOPMENT_TEAM` in `YOLO.xcodeproj/project.pbxproj `. Also add a +1 on the build number. Have you tested the branch independently?

File: YOLO.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -350,7 +330,7 @@
 				ASSETCATALOG_COMPILER_INCLUDE_ALL_APPICON_ASSETS = NO;
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 0;
-				DEVELOPMENT_TEAM = 3MR4P6CL3X;
+				DEVELOPMENT_TEAM = "";
```
Comment (ambitious-octopus) [inline YOLO.xcodeproj/project.pbxproj:333]:
```suggestion
				DEVELOPMENT_TEAM = 3MR4P6CL3X;
```

File: YOLO.xcodeproj/project.pbxproj (hunk 6 above)
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj]: I revert these reference.

**Comment by john-rocky** [comment]
I revert the model reference and increment the bundle version.
I tested this branch. This branch works fine.

**Review by ambitious-octopus — CHANGES_REQUESTED** [review]
Thank you @john-rocky . Let's fix the formatting and the PR is ready!

File: YOLO.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1,426 +1,426 @@
 // !$*UTF8*$!
 {
-	archiveVersion = 1;
-	classes = {
-	};
-	objectVersion = 54;
-	objects = {
+    archiveVersion = 1;
```
[the remark this answers is not in the document]
Comment (john-rocky) [reply YOLO.xcodeproj/project.pbxproj]: @ambitious-octopus 
I restored the indentation.

**Review by ambitious-octopus — COMMENTED** [review]
Thanks for resolving the issue, @john-rocky . @glenn-jocher , this PR is now ready and has been thoroughly tested by @john-rocky . In summary, it fixes the button for sharing photos with bounding boxes and also adds a preview of the image before it is sent.

**Comment by glenn-jocher** [comment]
@john-rocky @ambitious-octopus thanks guys, but I don't think we want an image preview. The image is already previewed in most sharing tools like WhatsApp or Slack before the user commits to sharing the image, so we're just inserting an additional step into their already complicated world.

Also I don't see any adaptation in the code for landscape, so the preview image will have the high gap top and bottom even in landscape. This is why we don't want to load this down with features and functions, this is a minimum viable product (MVP) that users can use to start building their own apps, we don't want this to be a fully feature product.

So please delete the preview and then this should be good to merge!

**Comment by john-rocky** [comment]
@glenn-jocher @ambitious-octopus 
To remove the preview, simply unmerge this branch.
If there are no problems, close this branch.

**Review by ambitious-octopus — COMMENTED** [review]
This PR is specifically for adding preview functionality. If we decide against this feature, we can close the PR. Does that sound right, @john-rocky ? This was implemented based on user feedback from those testing apps that currently lack a preview option.

**Comment by john-rocky** [comment]
@ambitious-octopus 
Yes. That's right.
